File: Python/1055-the-worlds_richest.py


class Billionaire():
    def __init__(self, name, age, net_worth):
        self.name = name
        self.age = age
        self.net_worth = net_worth

    def __lt__(self, other):
        if self.net_worth > other.net_worth:
            return True
        elif self.net_worth < other.net_worth:
            return False
        # if net_worth are equal
        if self.age < other.age:
            return True
        elif self.age > other.age:
            return False
        # if net_worth & age are equal
        return self.name < other.name

    def __str__(self):
        return f'{self.name} {self.age} {self.net_worth}'


if __name__ == '__main__':
    n_billionare, n_query = [int(i) for i in input().split()]
    database = []
    for _ in range(n_billionare):
        name, age, net_worth = input().split()
        database.append(Billionaire(name, int(age), int(net_worth)))
    database.sort()
    queries = []
    for i in range(1, n_query + 1):
        max_output, a_min, a_max = [int(i) for i in input().split()]
        count = 0
        print(f'Case #{i}:')
        for billionaire in database:
            if a_min <= billionaire.age <= a_max:
                print(billionaire)
                count += 1
            if count >= max_output:
                break
        if count == 0:
            print("None")

File: C++/1055-the-worlds_richest.cc
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>


struct Billionaire {
    std::string name;
    int age;
    int net_worth;    
};


bool comp(const Billionaire &s1, const Billionaire &s2) {
    if (s1.net_worth > s2.net_worth) {
        return true;
    } else if (s1.net_worth < s2.net_worth) {
        return false;
    }
    if (s1.age < s2.age) {
        return true;
    } else if (s1.age > s2.age) {
        return false;
    }
    return s1.name < s2.name;
}


int main() {

    std::ios::sync_with_stdio(false);

    int n_billionaire, n_query;
    std::cin >> n_billionaire >> n_query;

    std::vector<Billionaire> database;
    while (n_billionaire--) {
        std::string name;
        int age, net_worth;
        std::cin >> name >> age >> net_worth;
        database.push_back({name, age, net_worth});
    }

    std::sort(database.begin(), database.end(), comp);
    
    for (int i=1; i<=n_query; ++i) {
        int max_output, a_min, a_max;
        std::cin >> max_output >> a_min >> a_max;
        std::cout << "Case #" << i << ":\n";
        int count = 0;
        for (Billionaire &billionaire : database) {
            if (billionaire.age >= a_min && billionaire.age <= a_max) {
                std::cout << billionaire.name << ' '
                          << billionaire.age << ' '
                          << billionaire.net_worth << '\n';
                ++count;
            }
            if (count >= max_output) break;
        }
        if (count == 0) {
            std::cout << "None\n";
        }
    }

    return 0;
}
